include/mbgl/gfx/drawable.hpp

```diff
@@ -0,0 +1,230 @@
+
+    /// Drawable name is used for debugging and troubleshooting
+    const std::string& getName() const { return name; }
+    void setName(std::string value) { name = std::move(value); }
+
+    /// Which shader to use when rendering this drawable
+    const gfx::ShaderProgramBasePtr& getShader() const { return shader; }
+    void setShader(gfx::ShaderProgramBasePtr value) { shader = std::move(value); }
+
+    /// The pass on which we'll be rendered
+    mbgl::RenderPass getRenderPass() const { return renderPass; }
+    void setRenderPass(mbgl::RenderPass value) { renderPass = value; }
+
+    /// Test whether to draw this drawable in a given render pass.
+    bool hasRenderPass(const mbgl::RenderPass value) const {
+        return (mbgl::underlying_type(renderPass) & mbgl::underlying_type(value)) != 0;
+    }
+
+    /// Test whether to draw this drawable in a given render pass.
+    /// If multiple render pass bits are set, all must be present.
+    bool hasAllRenderPasses(const mbgl::RenderPass value) const {
+        const auto underlying_value = mbgl::underlying_type(value);
+        return (mbgl::underlying_type(renderPass) & underlying_value) == underlying_value;
+    }
+
+    /// Width for lines
+    int32_t getLineWidth() const { return lineWidth; }
+    void setLineWidth(int32_t value) { lineWidth = value; }
+
+    /// @brief Remove an attached texture from this drawable at the given sampler location
+    /// @param location Texture sampler location
+    void removeTexture(int32_t location);
+
+    /// @brief Return the textures attached to this drawable
+    /// @return Texture and sampler location pairs
+    const Textures& getTextures() const { return textures; };
+
+    /// @brief Get the texture at the given sampler location.
+    const gfx::Texture2DPtr& getTexture(int32_t location) const;
+
+    /// @brief Set the collection of textures bound to this drawable
+    /// @param textures_ A Textures collection to set
+    void setTextures(const Textures& textures_) noexcept { textures = textures_; }
+    void setTextures(Textures&& textures_) noexcept { textures = std::move(textures_); }
+
+    /// @brief Attach the given texture to this drawable at the given sampler location.
+    /// @param texture Texture2D instance
+    /// @param location A sampler location in the shader being used with this drawable.
+    void setTexture(gfx::Texture2DPtr texture, int32_t location);
+
+    /// Whether the drawble should be drawn
+    bool getEnabled() const { return enabled; }
+    void setEnabled(bool value) { enabled = value; }
+
+    /// Whether to render to the color target
+    bool getEnableColor() const { return enableColor; }
+    void setEnableColor(bool value) { enableColor = value; }
+
+    /// Whether to do stenciling (based on the Tile ID or 3D)
+    bool getEnableStencil() const { return enableStencil; }
+    void setEnableStencil(bool value) { enableStencil = value; }
+
+    /// not used for anything yet
```

Comment:
Is this still true?

Reply:
Seems to still be the case.

Reply:
Yes.  It's part of the comparison, but the only layer where ordering has been observed to matter (circle) is relying on the fallback ordering of assigned unique IDs.  We probably should be using explicit priority instead, as one could argue that we shouldn't assume monotonic increasing IDs.